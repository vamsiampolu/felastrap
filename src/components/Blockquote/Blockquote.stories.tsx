import type { Meta, StoryObj } from "@storybook/react";
import { Blockquote } from "../Blockquote/Blockquote";

export type Story = StoryObj<typeof Blockquote>;
export type Metadata = Meta<typeof Blockquote>;

const meta: Metadata = {
  title: "components/Blockquote",
  component: Blockquote,
  tags: ["autodocs"],
  parameters: {
    layout: ["centered"],
  },
};

export default meta;

export const Simple: Story = {
  args: {
    text: "Time cools, time clarifies; no mood can be maintained quite unaltered through the course of hours.",
  },
};

export const WithAttribution: Story = {
  args: {
    text: "Time cools, time clarifies; no mood can be maintained quite unaltered through the course of hours.",
    attribution: "Mark Twain",
  },
};

export const WithAttributionAndSource: Story = {
  args: {
    text: "It is better to be hated for what you are than to be loved for what you are not.",
    attribution: "Andre Gide",
    source: "Autumn Leaves",
  },
};

export const CenterAlign: Story = {
  args: {
    align: "center",
    text: "It is better to be hated for what you are than to be loved for what you are not.",
    attribution: "Andre Gide",
    source: "Autumn Leaves",
  },
};

export const RightAlign: Story = {
  args: {
    align: "right",
    text: "It is better to be hated for what you are than to be loved for what you are not.",
    attribution: "Andre Gide",
    source: "Autumn Leaves",
  },
};
