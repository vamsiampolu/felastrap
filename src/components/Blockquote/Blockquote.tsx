import type { ComponentPropsWithoutRef } from "react";
import { useFela } from "react-fela";

export type Align = "left" | "center" | "right";

export type BlockquoteProps = ComponentPropsWithoutRef<"blockquote"> & {
  text: string;
  attribution?: string;
  align?: Align;
  source?: string;
};

const rule = () => ({
  margin: "0 0 1rem",
  marginBottom: "1rem",
  fontSize: "1.25rem",
  "> :last-child": {
    marginBottom: 0,
  },
});

const captionRule = () => ({
  marginTop: "-1rem",
  marginBottom: "1rem",
  fontSize: "0.875em",
  color: "#6c757d",
  "::before": {
    content: '"- "',
  },
});

export const Blockquote = ({
  text,
  attribution = "",
  align = "left",
  className = "",
  source = "",
}: BlockquoteProps) => {
  const { css } = useFela({ align });
  const quoteClassName = `${css(rule)} ${className}`.trim();

  if (attribution) {
    return (
      <figure
        data-testid="blockquote-figure"
        className={css({ textAlign: align })}
      >
        <blockquote data-testid="blockquote" className={quoteClassName}>
          <p data-testid="blockquote-text">{text}</p>
        </blockquote>
        <figcaption
          data-testid="blockquote-attribution"
          className={css(captionRule)}
        >
          {attribution}
          {source ? (
            <>
              {" "}
              <cite data-testid="blockquote-source" title={source}>
                {source}
              </cite>
            </>
          ) : null}
        </figcaption>
      </figure>
    );
  }

  return (
    <blockquote data-testid="blockquote" className={quoteClassName}>
      <p>{text}</p>
    </blockquote>
  );
};
