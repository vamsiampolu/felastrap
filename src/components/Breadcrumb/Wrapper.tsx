import { useFela } from "react-fela";
import type { CSSProperties } from "fela";
import type { ComponentPropsWithoutRef } from "react";
import type { Theme, StyleProps } from "../common";

export type WrapperProps = ComponentPropsWithoutRef<"ol"> & {
  backgroundColor: string | null;
  borderRadius: string | number | null;
};

const rule = ({
  backgroundColor = null,
  borderRadius = null,
}: StyleProps<WrapperProps>) => {
  const padding = {
    paddingLeft: 0,
    paddingRight: 0,
    paddingTop: 0,
    paddingBottom: 0,
  };

  const style: CSSProperties = {
    ...padding,
    display: "flex",
    flexWrap: "wrap",
    marginBottom: "1rem",
    fontSize: "1rem",
    listStyle: "none",
  };

  if (backgroundColor) {
    style.backgroundColor = backgroundColor;
  }

  if (borderRadius) {
    style.borderRadius = borderRadius;
  }

  return style;
};

export const Wrapper = ({
  borderRadius = null,
  backgroundColor = null,
  children,
  ...other
}: WrapperProps) => {
  const { css } = useFela<Theme, WrapperProps>({
    borderRadius,
    backgroundColor,
  });

  const className = css(rule);
  return (
    <nav data-testid="breadcrumb" aria-label="Breadcrumb">
      <ol className={className} {...other}>
        {children}
      </ol>
    </nav>
  );
};
