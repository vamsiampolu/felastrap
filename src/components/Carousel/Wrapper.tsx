import { DivProps } from "../common";
import { useFela } from "react-fela";

export const Wrapper = ({ className = "", children, ...other }: DivProps) => {
  const { css } = useFela();
  const cx = `${css({ position: "relative" })} ${className} carousel`;
  return (
    <div data-testid="carousel" className={cx} {...other}>
      {children}
    </div>
  );
};
