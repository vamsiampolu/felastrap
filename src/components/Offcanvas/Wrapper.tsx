import { CSSCustomPrimitive, CSSProperties, IStyle } from "fela";
import { ElementType, forwardRef } from "react";
import { useFela } from "react-fela";
import {
  PolymorphicComponentPropsWithoutRef,
  PolymorphicRef,
} from "../../polymorphicComponentProps";
import { StyleProps, Theme } from "../common";

export type OffcanvasPlacement = "start" | "top" | "end" | "bottom";
export type CustomWrapperProps = {
  id: string;
  titleId: string;
  width?: CSSProperties["width"];
  isOpen?: boolean;
  placement?: OffcanvasPlacement;
};

type RuleProps = Omit<CustomWrapperProps, "id" | "titleId">;

export type WrapperProps<T extends ElementType> =
  PolymorphicComponentPropsWithoutRef<T, CustomWrapperProps>;

export type RuleResult = IStyle | CSSCustomPrimitive;

const rule = (): RuleResult => ({
  position: "fixed",
  bottom: 0,
  zIndex: 1045,
  display: "flex",
  flexDirection: "column",
  maxWidth: "100%",
  color: "#212529",
  visibility: "hidden",
  backgroundColor: "#fff",
  backgroundClip: "padding-box",
  outline: 0,
  transition: "transform 0.3s ease-in-out",
});

const startRule = ({ placement, width }: StyleProps<RuleProps>) =>
  placement === "start"
    ? {
        top: 0,
        left: 0,
        width: width ? width : "400px",
        borderRight: "1px solid rgba(0, 0, 0, 0.175)",
        transform: "translateX(-100%)",
      }
    : {};

const endRule = ({ placement, width }: StyleProps<RuleProps>): RuleResult =>
  placement === "end"
    ? {
        top: 0,
        right: 0,
        width: width ? width : "400px",
        borderLeft: "1px solid rgba(0, 0, 0, 0.175)",
        transform: "translateX(100%)",
      }
    : {};

const topRule = ({ placement }: StyleProps<RuleProps>): RuleResult =>
  placement === "top"
    ? {
        top: 0,
        right: 0,
        left: 0,
        height: "auto",
        maxHeight: "100%",
        borderBottom: "1px solid rgba(0, 0, 0, 0.175)",
        transform: "translateY(-100%)",
      }
    : {};

const bottomRule = ({ placement }: StyleProps<RuleProps>): RuleResult =>
  placement === "bottom"
    ? {
        right: 0,
        left: 0,
        height: "auto",
        maxHeight: "100%",
        borderTop: "1px solid rgba(0, 0, 0, 0.175)",
        transform: "translateY(100%)",
      }
    : {};

const showStartRule = ({
  placement,
  isOpen,
}: StyleProps<RuleProps>): RuleResult =>
  isOpen && placement === "start"
    ? {
        transform: "none",
        visibility: "visible",
      }
    : {
        transform: "translateX(-100%)",
        visibility: "hidden",
      };

const showEndRule = ({
  placement,
  isOpen,
}: StyleProps<RuleProps>): RuleResult =>
  isOpen && placement === "end"
    ? {
        transform: "none",
        visibility: "visible",
      }
    : {
        transform: "translateX(100%)",
        visibility: "hidden",
      };

const showTopRule = ({
  isOpen,
  placement,
}: StyleProps<RuleProps>): RuleResult =>
  isOpen && placement === "top"
    ? {
        transform: "none",
        visibility: "visible",
      }
    : {
        transform: "translateY(-100%)",
        visibility: "hidden",
      };

const showBottomRule = ({
  isOpen,
  placement,
}: StyleProps<RuleProps>): RuleResult =>
  isOpen && placement === "bottom"
    ? {
        transform: "none",
        visibility: "visible",
      }
    : {
        transform: "translateY(100%)",
        visibility: "hidden",
      };

const showPlacementRule = ({
  isOpen,
  placement,
}: StyleProps<RuleProps>): RuleResult => {
  switch (placement) {
    case "bottom": {
      return showBottomRule({
        isOpen,
        placement,
      } as StyleProps<RuleProps>);
    }

    case "end": {
      return showEndRule({
        isOpen,
        placement,
      } as StyleProps<RuleProps>);
    }

    case "top": {
      return showTopRule({ isOpen, placement } as StyleProps<RuleProps>);
    }

    case "start": {
      return showStartRule({
        isOpen,
        placement,
      } as StyleProps<RuleProps>);
    }
  }

  return {};
};

export const WrapperComponent = <T extends ElementType = "div">(
  {
    as,
    id,
    className,
    isOpen = false,
    titleId,
    children,
    placement = "start",
    width = undefined,
    ...props
  }: WrapperProps<T>,
  ref: PolymorphicRef<T>,
) => {
  const { css } = useFela<Theme, RuleProps>({ isOpen, placement, width });
  const cx = `${css([
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    rule as any,
    startRule,
    endRule,
    topRule,
    bottomRule,
    showPlacementRule,
  ])} offcanvas offcanvas-${placement} ${className || ""}`;
  const Component = as || "div";
  return (
    <Component
      data-testid="offcanvas-wrapper"
      ref={ref}
      className={cx}
      tabIndex={-1}
      id={id}
      aria-labelledby={titleId}
      {...props}
    >
      {children}
    </Component>
  );
};

export const Wrapper = forwardRef(WrapperComponent);
Wrapper.displayName = "Wrapper";
