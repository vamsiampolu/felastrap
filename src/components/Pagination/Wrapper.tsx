import { ComponentPropsWithoutRef } from "react";
import { useFela } from "react-fela";

export type WrapperProps = ComponentPropsWithoutRef<"ul"> &
  Partial<{ size: number }>;

export const Wrapper = ({
  className = "",
  children,
  ...other
}: WrapperProps) => {
  const { css } = useFela();
  const style = {
    display: "flex",
    listStyle: "none",
    paddingLeft: 0,
  };
  const label = other["aria-label"] || "Page Navigation";
  const cx = `${css(style)} ${className}`;
  return (
    <nav data-testid="pagination-wrapper" aria-label={label}>
      <ul className={cx}>{children}</ul>
    </nav>
  );
};
