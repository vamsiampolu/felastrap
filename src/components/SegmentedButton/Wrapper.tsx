import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  position: "relative",
  display: "flex",
  flexWrap: "wrap",
  alignItems: "stretch",
  width: "100%",
  marginBottom: "1rem",
});

export const Wrapper = ({ className = "", children, ...props }: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} segmented-btn-wrapper ${className}`.trim();
  return (
    <div data-testid="segmented-button" className={cx} {...props}>
      {children}
    </div>
  );
};
