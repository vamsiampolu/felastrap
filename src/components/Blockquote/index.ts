/* istanbul ignore file */
import { Blockquote, type BlockquoteProps } from "./Blockquote";
export { Blockquote, type BlockquoteProps };
