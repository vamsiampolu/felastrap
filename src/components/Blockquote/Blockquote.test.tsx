import { IRenderer } from "fela";
import { Wrapper } from "../../test-helpers/Wrapper";
import { Align, Blockquote } from "./Blockquote";
import { createRenderer, cssReboot } from "../../test-helpers/createRenderer";
import { render, screen } from "@testing-library/react";
import { createSnapshot } from "../../test-helpers/createSnapshot";

describe("components/Blockquote", () => {
  let renderer: IRenderer;

  beforeAll(() => {
    renderer = createRenderer();
  });

  beforeEach(() => {
    renderer.clear();
    cssReboot(renderer);
  });

  afterAll(() => {
    renderer = undefined as unknown as IRenderer;
  });

  it("should render a quote", () => {
    const component = (
      <Wrapper renderer={renderer}>
        <Blockquote text="Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at." />
      </Wrapper>
    );

    render(component);
    expect(screen.getByTestId("blockquote")).toBeInTheDocument();
    expect(screen.queryByTestId("blockquote")).toHaveTextContent(
      "Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at.",
    );
  });

  it("should render a quote with attribution", () => {
    const component = (
      <Wrapper renderer={renderer}>
        <Blockquote
          attribution="Stephen Hawking"
          text="Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at."
        />
      </Wrapper>
    );

    render(component);

    expect(screen.getByTestId("blockquote-figure")).toBeInTheDocument();
    expect(screen.getByTestId("blockquote")).toBeInTheDocument();
    expect(screen.getByTestId("blockquote-text")).toBeInTheDocument();

    expect(screen.queryByTestId("blockquote-text")).toHaveTextContent(
      "Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at.",
    );

    expect(screen.getByTestId("blockquote-attribution")).toBeInTheDocument();
    expect(screen.queryByTestId("blockquote-attribution")).toHaveTextContent(
      "Stephen Hawking",
    );
  });

  it("should render a quote with attribution and source", () => {
    const component = (
      <Wrapper renderer={renderer}>
        <Blockquote
          source="Reith Lecture (BBC)"
          attribution="Stephen Hawking"
          text="Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at."
        />
      </Wrapper>
    );

    render(component);

    expect(screen.getByTestId("blockquote-text")).toBeInTheDocument();
    expect(screen.queryByTestId("blockquote-text")).toHaveTextContent(
      "Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at.",
    );

    expect(screen.queryByTestId("blockquote-attribution")).toHaveTextContent(
      "Stephen Hawking",
    );

    const source = screen.queryByTestId("blockquote-source");

    expect(screen.getByTestId("blockquote-source")).toBeInTheDocument();
    expect(source).toHaveAttribute("title", "Reith Lecture (BBC)");
    expect(source).toHaveTextContent("Reith Lecture (BBC)");
  });

  describe("styling", () => {
    it.each(["left", "center", "right"])(
      "should render a quote %s aligned",
      async (align) => {
        const component = (
          <Blockquote
            align={align as Align}
            text="Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at."
          />
        );
        const snapshot = await createSnapshot({
          component,
          renderer,
          theme: {},
        });
        expect(snapshot).toMatchSnapshot();
      },
    );

    it.each(["left", "center", "right"])(
      "should render a quote with attribution %s aligned",
      async (align) => {
        const component = (
          <Blockquote
            align={align as Align}
            attribution="Stephen Hawking"
            text="Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at."
          />
        );
        const snapshot = await createSnapshot({
          component,
          renderer,
          theme: {},
        });
        expect(snapshot).toMatchSnapshot();
      },
    );

    it.each(["left", "center", "right"])(
      "should render a quote with attribution and source",
      async (align) => {
        const component = (
          <Blockquote
            align={align as Align}
            source="Reith Lecture (BBC)"
            attribution="Stephen Hawking"
            text="Remember to look up at the stars and not down at your feet. Try to make sense of what you see and wonder about what makes the universe exist. Be curious. And however difficult life may seem, there is always something you can do and succeed at."
          />
        );
        const snapshot = await createSnapshot({
          component,
          renderer,
          theme: {},
        });
        expect(snapshot).toMatchSnapshot();
      },
    );
  });
});
