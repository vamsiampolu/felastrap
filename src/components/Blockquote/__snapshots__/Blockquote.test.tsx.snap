// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Blockquote styling should render a quote center aligned 1`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}


<blockquote data-testid="blockquote" class="a b c d e f">
  <p>
    Remember to look up at the stars and not down at your feet. Try to make
    sense of what you see and wonder about what makes the universe exist. Be
    curious. And however difficult life may seem, there is always something you
    can do and succeed at.
  </p>
</blockquote>
;
"
`;

exports[`components/Blockquote styling should render a quote left aligned 1`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}


<blockquote data-testid="blockquote" class="a b c d e f">
  <p>
    Remember to look up at the stars and not down at your feet. Try to make
    sense of what you see and wonder about what makes the universe exist. Be
    curious. And however difficult life may seem, there is always something you
    can do and succeed at.
  </p>
</blockquote>
;
"
`;

exports[`components/Blockquote styling should render a quote right aligned 1`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}


<blockquote data-testid="blockquote" class="a b c d e f">
  <p>
    Remember to look up at the stars and not down at your feet. Try to make
    sense of what you see and wonder about what makes the universe exist. Be
    curious. And however difficult life may seem, there is always something you
    can do and succeed at.
  </p>
</blockquote>
;
"
`;

exports[`components/Blockquote styling should render a quote with attribution and source 1`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}
.g {
  text-align: left;
}
.h {
  margin-top: -1rem;
}
.i {
  font-size: 0.875em;
}
.j {
  color: #6c757d;
}
.k::before {
  content: "- ";
}


<figure data-testid="blockquote-figure" class="g">
  <blockquote data-testid="blockquote" class="a b c d e f">
    <p data-testid="blockquote-text">
      Remember to look up at the stars and not down at your feet. Try to make
      sense of what you see and wonder about what makes the universe exist. Be
      curious. And however difficult life may seem, there is always something
      you can do and succeed at.
    </p>
  </blockquote>
  <figcaption data-testid="blockquote-attribution" class="a h i j k">
    Stephen Hawking
    <cite data-testid="blockquote-source" title="Reith Lecture (BBC)"
      >Reith Lecture (BBC)</cite
    >
  </figcaption>
</figure>
;
"
`;

exports[`components/Blockquote styling should render a quote with attribution and source 2`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}
.g {
  text-align: center;
}
.h {
  margin-top: -1rem;
}
.i {
  font-size: 0.875em;
}
.j {
  color: #6c757d;
}
.k::before {
  content: "- ";
}


<figure data-testid="blockquote-figure" class="g">
  <blockquote data-testid="blockquote" class="a b c d e f">
    <p data-testid="blockquote-text">
      Remember to look up at the stars and not down at your feet. Try to make
      sense of what you see and wonder about what makes the universe exist. Be
      curious. And however difficult life may seem, there is always something
      you can do and succeed at.
    </p>
  </blockquote>
  <figcaption data-testid="blockquote-attribution" class="a h i j k">
    Stephen Hawking
    <cite data-testid="blockquote-source" title="Reith Lecture (BBC)"
      >Reith Lecture (BBC)</cite
    >
  </figcaption>
</figure>
;
"
`;

exports[`components/Blockquote styling should render a quote with attribution and source 3`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}
.g {
  text-align: right;
}
.h {
  margin-top: -1rem;
}
.i {
  font-size: 0.875em;
}
.j {
  color: #6c757d;
}
.k::before {
  content: "- ";
}


<figure data-testid="blockquote-figure" class="g">
  <blockquote data-testid="blockquote" class="a b c d e f">
    <p data-testid="blockquote-text">
      Remember to look up at the stars and not down at your feet. Try to make
      sense of what you see and wonder about what makes the universe exist. Be
      curious. And however difficult life may seem, there is always something
      you can do and succeed at.
    </p>
  </blockquote>
  <figcaption data-testid="blockquote-attribution" class="a h i j k">
    Stephen Hawking
    <cite data-testid="blockquote-source" title="Reith Lecture (BBC)"
      >Reith Lecture (BBC)</cite
    >
  </figcaption>
</figure>
;
"
`;

exports[`components/Blockquote styling should render a quote with attribution center aligned 1`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}
.g {
  text-align: center;
}
.h {
  margin-top: -1rem;
}
.i {
  font-size: 0.875em;
}
.j {
  color: #6c757d;
}
.k::before {
  content: "- ";
}


<figure data-testid="blockquote-figure" class="g">
  <blockquote data-testid="blockquote" class="a b c d e f">
    <p data-testid="blockquote-text">
      Remember to look up at the stars and not down at your feet. Try to make
      sense of what you see and wonder about what makes the universe exist. Be
      curious. And however difficult life may seem, there is always something
      you can do and succeed at.
    </p>
  </blockquote>
  <figcaption data-testid="blockquote-attribution" class="a h i j k">
    Stephen Hawking
  </figcaption>
</figure>
;
"
`;

exports[`components/Blockquote styling should render a quote with attribution left aligned 1`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}
.g {
  text-align: left;
}
.h {
  margin-top: -1rem;
}
.i {
  font-size: 0.875em;
}
.j {
  color: #6c757d;
}
.k::before {
  content: "- ";
}


<figure data-testid="blockquote-figure" class="g">
  <blockquote data-testid="blockquote" class="a b c d e f">
    <p data-testid="blockquote-text">
      Remember to look up at the stars and not down at your feet. Try to make
      sense of what you see and wonder about what makes the universe exist. Be
      curious. And however difficult life may seem, there is always something
      you can do and succeed at.
    </p>
  </blockquote>
  <figcaption data-testid="blockquote-attribution" class="a h i j k">
    Stephen Hawking
  </figcaption>
</figure>
;
"
`;

exports[`components/Blockquote styling should render a quote with attribution right aligned 1`] = `
".a {
  margin-bottom: 1rem;
}
.b {
  font-size: 1.25rem;
}
.c > :last-child {
  margin-bottom: 0;
}
.d {
  margin-top: 0;
}
.e {
  margin-right: 0;
}
.f {
  margin-left: 0;
}
.g {
  text-align: right;
}
.h {
  margin-top: -1rem;
}
.i {
  font-size: 0.875em;
}
.j {
  color: #6c757d;
}
.k::before {
  content: "- ";
}


<figure data-testid="blockquote-figure" class="g">
  <blockquote data-testid="blockquote" class="a b c d e f">
    <p data-testid="blockquote-text">
      Remember to look up at the stars and not down at your feet. Try to make
      sense of what you see and wonder about what makes the universe exist. Be
      curious. And however difficult life may seem, there is always something
      you can do and succeed at.
    </p>
  </blockquote>
  <figcaption data-testid="blockquote-attribution" class="a h i j k">
    Stephen Hawking
  </figcaption>
</figure>
;
"
`;
